### PR TITLE
GTJ-25 reg.front.end

### PR DESCRIPTION
**Pull request by zvorsi**

### CODE REVIEW

File: frontend/src/components/Registration.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    constructor(props) {
+        super(props);
+        // console.log(props)
+        this.state = {
+            newUser: {
+              username: '',
+              password: '',
+              kingdomname: ''
+            }
+          };
```
Comment (adamgyulavari) [inline frontend/src/components/Registration.js:19]: @zvorsi this still doesnt have to be a separate newUser property on the state:
```suggestion
        this.state = {
              username: '',
              password: '',
              kingdomname: ''
          };

```

File: frontend/src/components/Registration.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+            <div className="form">
+                <form 
+               
+                onSubmit = {this.handleSubmit}>
+                    <div>
+                    <input type="text" name="username" className={this.validUsername() ? "green" : "red"} placeholder="Username" onChange = {this.handleUsername}>
+                    </input>
+                    <div className={this.validUsername() ? "" : "redimage"}><span className={this.validUsername() ? "alertnotext" : ""}>Username must be at least 1 character long!</span></div>
+                    </div>
```
Comment (adamgyulavari) [inline frontend/src/components/Registration.js:110]: @zvorsi if it's valid you can just leave out the whole div (and the same goes for the others)
```suggestion
                    {!this.validUsername() && <div className="redimage"><span>Username must be at least 1 character long!</span></div>}
                    </div>

```

File: frontend/src/components/Registration.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    initialEntries: ['/']
+  });
+
+
+let utils =  ()=>{return render(
+    <Router history={history}>
+        <Route path = '/'>
+            <Registration />
+        </Route>
+    </Router>
```
Comment (adamgyulavari) [inline frontend/src/components/Registration.test.js:17]: @zvorsi this is not really necessary to put it into the router, you can just render a `Registration` component to test

Comment (zvorsi) [reply frontend/src/components/Registration.test.js:17]: at first it passes, then running test again fails, I would keep it that way

File: frontend/src/components/Registration.js [hunk 4]
```diff
@@ -46,7 +42,7 @@ class Registration extends Component {
                   if(data.error){
                     alert(data.error)
                   }else{
-                    alert("welcome " + data.username)
+                    // alert("welcome " + data)
```
Comment (adamgyulavari) [inline frontend/src/components/Registration.js:45]: @zvorsi let's remove the `alert` and `console.log` comments :)

File: frontend/src/components/Registration.js [hunk 5]
```diff
@@ -55,39 +51,39 @@ class Registration extends Component {
 
         handleUsername(e){
           let value = e.target.value;
-          this.setState( prevState => ({ newUser :
-            {...prevState.newUser, username: value
-            }
+          this.setState( prevState => (
+            {...prevState.username, username: value
```
Comment (adamgyulavari) [inline frontend/src/components/Registration.js:55]: @zvorsi this is not going to be good. try it out and fix it

File: frontend/src/components/Registration.js [hunk 6]
```diff
@@ -40,36 +39,35 @@ class Registration extends Component {
                 ).then(data => {
                   
                   if(data.error){
-                    alert(data.error)
+                    this.setState(prevState => (
+                      {...prevState, error: data.error}
+                    ))
+                    // history.push('/registration');
```
Comment (adamgyulavari) [inline frontend/src/components/Registration.js:45]: @zvorsi remove comment

File: frontend/src/components/Registration.js [hunk 7]
```diff
@@ -102,23 +100,26 @@ class Registration extends Component {
                     <div>
                     <input type="text" name="username" className={this.validUsername() ? "green" : "red"} placeholder="Username" onChange = {this.handleUsername}>
                     </input>
-                    {!this.validUsername() && <div className="redimage"><span>Username must be at least 1 character long!</span></div>}
+                    {/* {!this.validUsername() && <div className="redimage"><span>Username must be at least 1 character long!</span></div>} */}
```
Comment (adamgyulavari) [inline frontend/src/components/Registration.js:103]: @zvorsi remove comment, and this should not be a comment actually

Comment (adamgyulavari) [reply frontend/src/components/Registration.js:103]: same goes for the other input fields

**Pull request merged**